client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -193,12 +206,48 @@ private void logFailedRequest(RequestBuilder fullReq, Throwable throwable) {
             .flatMap(str -> deserialize(method, str));
     }
 
-    private Observable<HttpClientResponse<ByteBuf>> logEmptyResponse(RequestBuilder request) {
+    /**
+     * Retries the request if not a post.
+     *
+     * @param request      original request
+     * @param method       original method
+     * @param arguments    originial argument
+     * @param rxClient     original rxClient
+     * @param retryIfFails true if we should retry if this requests fails as well. False if we should not retry anymore
+     * @return Observable of HttpClientResponse
+     */
+    private Observable<HttpClientResponse<ByteBuf>> retry(RequestBuilder request,
+        Method method,
+        Object[] arguments,
+        io.reactivex.netty.protocol.http.client.HttpClient<ByteBuf, ByteBuf> rxClient,
+        boolean retryIfFails
+    ) {
         return defer(() -> {
 
-            LOG.warn(request + " with headers " + request.getHeaders().entrySet() + " got empty response from submit");
+            // Don't retry if it was a POST, as it is not idempotent and the request
+            // might have ended up at the server
+            if (request.getHttpMethod().equals(HttpMethod.POST)) {
+                LOG.info("Won't retry POST on {} ", request.getUri());
+                return empty();
+            }
+
+            RequestBuilder fullReq = getRequestBuilder(method, arguments);
```

Comment:
Why create it again? You got it as a paramater to the method?

Reply:
Yes, thought at first it may hold some state and to avoid this problem i created a new instance. But will try reusing the old one.